File: app/src/main/java/com/chatserver/dd/chat/Vip/View/ActivityVip.java
package com.chatserver.dd.chat.Vip.View;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import com.chatserver.dd.chat.R;
import com.chatserver.dd.chat.Vip.Model.Model;
import com.chatserver.dd.chat.Vip.Presenter.Presenter;

import java.util.List;

public class ActivityVip extends AppCompatActivity implements IView {

    private RecyclerView mRecyclerViewMain;
    private Presenter presenterAllWords;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_vip);
        initView();

        presenterAllWords = new Presenter(this,new Model());
        presenterAllWords.requestDataFromServer();

        getUserList();
    }

    private void getUserList() {

    }

    RecyclerItemClickListener recyclerItemClickListener = new RecyclerItemClickListener() {
        @Override
        public void onItemClick(UsersVip usersAllWords) {
            Toast.makeText(ActivityVip.this,
                    "Message:  " + usersAllWords.getMessage(),
                    Toast.LENGTH_LONG).show();
        }
    };

    private void initView() {

        mRecyclerViewMain = (RecyclerView) findViewById(R.id.recyclerViewMain);
        Toast.makeText(this, "", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void setDataToRecyclerView(List<UsersVip> userList) {
        LinearLayoutManager layoutManager = new LinearLayoutManager(ActivityVip.this);
        mRecyclerViewMain.setLayoutManager(layoutManager);
        Adapter recyclerViewAdapter = new Adapter(userList,recyclerItemClickListener);
        mRecyclerViewMain.setAdapter(recyclerViewAdapter);
    }

    @Override
    public void onResponseFailure(Throwable throwable) {

    }


}

File: app/src/main/java/com/chatserver/dd/chat/blocks/vip/ActivityVip.kt
package com.chatserver.dd.chat.blocks.vip

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.Serverserver.dd.Server.blocks.Server.AdapterVip
import com.chatserver.dd.chat.R
import com.chatserver.dd.chat.model.Vip
import kotlinx.android.synthetic.main.activity_chat.*


class ActivityVip : AppCompatActivity() {

    private lateinit var viewModel: ViewModelVipActivity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_vip)

        showToast("Started")
        viewModel = ViewModelProvider(this).get(ViewModelVipActivity::class.java)
        getData()
    }

    private fun getData() {
        viewModel.liveData.observe(this, Observer {
            setDataToRecyclerView(it)
        })
    }

    private fun showToast(text: String) {
        Toast.makeText(this, text, Toast.LENGTH_SHORT).show()
    }

    fun setDataToRecyclerView(list: List<Vip>) {
        recyclerViewMain.apply {
            layoutManager = LinearLayoutManager(this@ActivityVip)
            adapter = AdapterVip(list)
        }
    }
}
